### ISSUE TITLE
Print order error on magento 2.1.8

### ISSUE THREAD

**Issue by ionutdicu** [magento/magento2#10530]
## Preconditions
1. Magento 2.1.9

## Steps to Reproduce
1. Login as customer
2. Place order
3. Go to "My account" page on frontend
4. Click on your order
5. Click "Print Order"

## Actual Result
```
Exception (Magento\Framework\Exception\LocalizedException): Invalid method Magento\Sales\Block\Order\PrintShipment::isPagerDisplayed<br>
#0 \vendor\magento\module-sales\view\frontend\templates\order\items.phtml(15): Magento\Framework\DataObject->__call('isPagerDisplaye...', Array)
```

## Expected result
1. Order is printable

**Comment by pboisvert** [comment]
Can you please follow the reporting [template ](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)in terms of providing pre-conditions (configurations other than 2.1.8) , steps to reproduce when reporting an issue.

**Comment by ionutdicu** [comment]
This exception appear on frontend customer Dashboard(Recent Orders->Order View->Print Order) after upgrade magento 2.1.7 at 2.1.8.

**Comment by vityakopin** [comment]
Unfortunately, I could not reproduce the issue as you described it. Please add more details to your description of the steps you followed when identifying this issue. In addition, identify the web server you are running, the versions of PHP and MySQL, and any other information needed to reproduce your issue.

**Comment by tureki** [comment]
I think M2 sale module given wrong block class for `sales_order_print.xml`

Temporary way:
open [module-sale] `view -> frontend -> layout -> sales_order_print.xml`. then modify line 20

```xml
<block class="Magento\Sales\Block\Order\PrintShipment" name="order_items" template="order/items.phtml">
```
to
```xml
<block class="Magento\Sales\Block\Order\Items" name="order_items" template="order/items.phtml">
```

Then it will be fine.

---

If u r using custom theme just override the `layout` file to hotfix it. (no need change M2 source code)

**Comment by gigadesign1** [comment]
I'm having the same issue in Magento 2.1.8 and Magento 2.1.9.

In the layout.xml the block is defined as:
`<block class="Magento\Sales\Block\Order\PrintShipment" name="order_items" template="order/items.phtml">`

In the items.phtml template it states:
`/** @var \Magento\Sales\Block\Order\Items $block */`

Clearly, this is incorrect.

The template calls for
`$block->isPagerDisplayed()`

This function is not defined in Magento\Sales\Block\Order\PrintShipment, but is defined in Magento\Sales\Block\Order\Items

**Comment by grizwako** [comment]
I am having the same issue on Magento 2.1.8.

Steps to reproduce:
Login as customer
Place order
Go to your customer page on frontend
Click on your order
Click "Print Order"

**Comment by magento-engcom-team** [comment]
Please follow the issue reporting [template](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) for future reports.

**Comment by grizwako** [comment]
I am having the same issue on Magento 2.1.8.

Preconditions:
Magento 2.1.8 installed
Created configurable products manually in admin.


Steps to reproduce:
Login as customer
Place order using one of created products
Go to your customer page on frontend
Click on your order
Click "Print Order"

Expected Results:
Seeing "order details" in a print friendly render

Actual Results:
Exception (Magento\Framework\Exception\LocalizedException): Invalid method Magento\Sales\Block\Order\PrintShipment::isPagerDisplayed

**Comment by magento-engcom-team** [comment]
@ionutdicu We have updated issue report to follow guidelines.

**Comment by magento-engcom-team** [comment]
@ionutdicu, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.

**Comment by santhoshnsscoe** [comment]
We are also facing the same problem after update to 2.1.9. 

If development mode is enabled then, on Print Order, giving the below error.
```
1 exception(s):
Exception #0 (Magento\Framework\Exception\LocalizedException): Invalid method Magento\Sales\Block\Order\PrintShipment::isPagerDisplayed

Exception #0 (Magento\Framework\Exception\LocalizedException): Invalid method Magento\Sales\Block\Order\PrintShipment::isPagerDisplayed
#0 vendor/magento/module-sales/view/frontend/templates/order/items.phtml(15): Magento\Framework\DataObject->__call('isPagerDisplaye...', Array)
```

If production mode is enabled, then the items are missing in the print out. Below is the screenshot.
https://www.screencast.com/t/Trv5bZhQvl0q

**Comment by okorshenko** [comment]
The issue has been fixed in 2.1-develop branch and will be available with next patch release

**Comment by sidolov** [comment]
Hi @ionutdicu. Thank you for your report. 
The issue has been fixed in magento/magento2#17998 by @MateuszChrapek in 2.2-develop branch
Related commit(s):
 - [9316bd0eae2acc59dcceae09d212836083e93167](https://github.com/magento/magento2/commit/9316bd0eae2acc59dcceae09d212836083e93167)
 - [aaa55287f4be6f2c210d8026b8114f60e8735258](https://github.com/magento/magento2/commit/aaa55287f4be6f2c210d8026b8114f60e8735258)

The fix will be available with the upcoming 2.2.8 release.

**Comment by slavvka** [comment]
Hi @ionutdicu. Thank you for your report. 
The issue has been fixed in magento/magento2#18272 by @mage2pratik in 2.3-develop branch
Related commit(s):
 - [37ff9e94589847f505925add1ed74b6c65a009e2](https://github.com/magento/magento2/commit/37ff9e94589847f505925add1ed74b6c65a009e2)

The fix will be available with the upcoming 2.3.1 release.
